mypy/test/testcheck.py

```diff
@@ -154,6 +154,8 @@ def run_case_once(self, testcase: DataDrivenTestCase, incremental_step: int = 0)
                             # change. We manually set the mtime to circumvent this.
                             new_time = os.stat(target).st_mtime + 1
                             os.utime(target, times=(new_time, new_time))
+                for path in testcase.deleted_paths.get(incremental_step, set()):
+                    os.remove(path)
```

Comment:
I think you'll need a try/except around this similar to other places (see tear_down() in same file). That should prevent the AppVeyor flakes (in general on Windows remove() sometimes fails due to e.g. virus checkers having the file open).